### ISSUE TITLE
device handler exception error

### ISSUE THREAD

**Issue by hawk47** [GaryMilne/Tasmota-RGBCCT-DH-for-SmartThings-Classic-with-MQTT#2]
Hello, 
Than you for making your work available to all of us.
I receive a 'Org.hibernate.exception.GenericJDBCException: Could not execute JDBC batch update'
message when trying to create this device handler.
any ideas on what I might be doing wrong
thank you,
Steve

**Comment by GaryMilne** [comment]
Unfortunately that is a backend error that I don't have any control over.  There are a number of errors related to JDBC mentioned in the SmartThings community. Perhaps you will find something there to give you a clue. https://community.smartthings.com/search?q=JDBC

The good news is that they seem to be temporary issues related to the platform.  Sorry that I can't offer you some better guidance.

**Comment by hawk47** [comment]
Thank you for the response. I will keep trying.
Merry Christmas!

**Comment by jbeck22** [comment]
I'm getting the same error.  Any update on this?

**Comment by GaryMilne** [comment]
Just a few minutes ago I copied and pasted the code from Github into a new device driver and it worked O.K. for me.  Why two people would get different results I have no idea.  What you could try is creating a new device handler using some simple generic code. Once it is created you could then try "updating" it with my DH code.

Wish I could offer more but it is a platform error and I don't have access, nor can I replicate the error.
I'll contact hawk47 and see if his eventually worked.

**Comment by GaryMilne** [comment]
hawk47  Did this eventually work for you after multiple retries?

**Comment by hawk47** [comment]
still no luck here I tried modifying another DH code - no luck.
I am able to create/publish other DHs.

**Comment by GaryMilne** [comment]
hawk47, I don't get that error so I don't have enough info to try and figure it out.  But if I did have that error I would paste in smaller parts of the DH at a time to try and narrow down the issue.  I would start with omitting the contents of the log and message functions to see if the issue is related to the emoji characters. Just a guess.

**Comment by MatthieuCt** [comment]
Hi,
By removing parts of the devices handler, I figured out the error came from Emojis in code.
If you replace them with text, it will work !
You can find them in ```log``` and ```message``` functions.

**Comment by GaryMilne** [comment]
Hey Matthieu, I suspected it may be related to the emojis and just posted yesterday to that effect. Now that you have been able to create the DH would you mind trying to post in the original log and message functions (or parts thereof) and see if it will accept it as an update vs a brand new DH.

As I’ve mentioned it works for me as is and I do think the emojis add something to the overall display.

**Comment by jbeck22** [comment]
so you mean these?  If so, what did you put in place of the emojis?

<img width="863" alt="Screen Shot 2020-01-05 at 5 50 06 PM" src="https://user-images.githubusercontent.com/4844115/71787285-dc666080-2fe3-11ea-92f4-6abf2553dab3.png">

**Comment by GaryMilne** [comment]
jbeck22, yes, those are what is being referred to.  Try pasting in those functions after the DH has been created, log() and message() functions that is. If you can’t paste in the whole thing do it in parts.

**Comment by jbeck22** [comment]
Are you able to post your working code either on Git or somewhere else I can get to?  Would really like to try this, but still can’t get this to work.

Thank you,

Jeramey Beck

**Comment by GaryMilne** [comment]
Yes, those commands do need to be in there in order for it to work.  The code here on github is xactly the code I’m using, I just don’t get the error. Perhaps because I have several device handlers that also use emojis.

Now you have created the DH you can try pasting the log and message functions back in. Don’t know if that will work but it’s worth a try.

**Comment by fireboy1919** [comment]
You know, if you put them into the smartthings format, then people can just subscribe to your repository rather than copying and pasting.

That should make these errors not happen anymore.

Example of using the format:
https://github.com/fireboy1919/MiThings

This lets you use the "update from repo" command within smartthings hub.
